doc/cabal-project.rst

```diff
@@ -269,6 +269,19 @@ package, and thus apply globally:
     ``-j2`` is also supported; a bare ``--jobs`` or ``-j`` is equivalent
     to ``--jobs=$ncpus``.
 
+.. cfg-field::  semaphore: boolean
+                --semaphore
+    :synopsis: Use GHC's support for semaphore based parallelism.
+
+    :default: False
+
+    When combined with ``jobs``, this option instructs cabal to create a
+    semaphore with the number of slots specified by ``jobs``. This semaphore is
+    then passed to GHC using the `-jsem` option to make use of GHC's ability to
+    control parrelism using a semaphore.
```

Comment:
```suggestion
    This option instructs cabal to control parallelism by creating a new system semaphore,
    whose number of tokens is specified by ``--jobs`` (or ``-j``).
    This semaphore is passed to GHC, which allows it to use any leftover parallelism
    that ``cabal-install`` is not using.
    
    Requires ``ghc >= 9.8``.
```

Question: what happens if no ``--jobs`` argument is passed? Do we use `$ncpus`, or throw an error?

Reply:
By default `jobs: $ncpus`, but if by default the flag wasn't set then we should create a semaphore with one slot which would just behave like `-j1` normally would.